docs/porting/target/tickless/tickless.md

```diff
@@ -0,0 +1,64 @@
+
+[Mbed OS sleep modes](../../../apis/power-management-sleep.html):
+
+- Sleep mode: Both low and high-frequency clocks are active.
+- Deep sleep mode: Only the low-frequency clock is active.
+
+To enable tickless mode in Mbed OS, your target has to meet following requirements:
+
+- Support for sleep HAL API
+- Support for low power or microsecond ticker HAL API
+
+If the target supports both low power and microsecond ticker, then low power ticker is selected with tickless mode.
+
+For instance, the system runs only with two threads thread 1 and thread 2. During the first 2 tick interrupts thread 1 and thread 2 schedules alternatively. When thread 1 entered into a wait state, the scheduler preempts and schedules thread 2 for execution. After some time, thread 2 enters into a wait state and get preempted. Now scheduler schedules idle thread to run as two threads are in the wait state. This idle thread put the system to sleep. After multiple SysTick periodic interrupts, thread 2 is rescheduled for execution. All previous Systick wake-ups waste power.
+
+![](./resources/Normal_Tick.png)
+
+But with tickless mode, it suspends the SysTick periodic wake up as no threads are ready to run for quite some time which inturn saves the power.
+
+![](./resources/Tickless.png)
+
+### Enable tickless mode support in full profile
+
+Find your target in `mbed-os/targets/targets.json` and update its configuration as described below.
+
+1. Add `MBED_TICKLESS` macro into the `macros` option as follows:
+
+    ```json
+	"macros": [
+        "MBED_TICKLESS"
+    ]    
+    ```
+1. If your target does not have a low power ticker or you prefer to use microsecond ticker instead of low power ticker for tickless mode, then ensure that tickless is enabled using the microsecond ticker:
+
+    ```json
+    "overrides": {
+        "tickless-from-us-ticker": true
+    }
+    ```
+
+Targets supporting tickless mode override the default SysTick mechanism and use the [OsTimer](../mbed-os-api-doxy/structos__timer__def.html) implementation based on either the [low power ticker](../mbed-os-api-doxy/group__hal__lp__ticker.html) or [microsecond ticker](../mbed-os-api-doxy/group__hal__us__ticker.html) if "tickless-from-us-ticker" target configuration is set. This change is necessary to avoid drift connected with using two different timers to measure time. Users must not change the ticker interrupt handler when tickless mode is in use.
+
+### Enable tickless mode support in bare metal profile 
+
+Tickless mode is enabled by default for bare metal profile without adding `MBED_TICKLESS` target configuration. As mentioned earlier target prefer to use microsecond ticker for tickless mode or does not have low power ticker then add the below target configuration in `targets.json`.
```

Comment:
```suggestion
Tickless mode is enabled by default for bare metal profile without adding `MBED_TICKLESS` target configuration. As mentioned earlier, if you prefer to use microsecond ticker for tickless mode, or the target does not have a low power ticker, then add the below target configuration in `targets.json`.
```

Reply:
As described above, Tickless means "pausing" the SysTick, sleep for a duration, then resume the SysTick. But if the 1ms-SysTick is for thread scheduling, is it still enabled on bare-metal? How do we say "Tickless mode is enabled by default for bare metal profile"?

Reply:
You are correct. SysTick is never used in bare metal.
But timed sleeping RTOS API calls like _Semaphore::try_acquire_for_ or _ThisThread::sleep_until_, it directly requests a wake-up from the SysTimer and calls sleep.

Reply:
@LDong-Arm is exactly rigth. It is not right to say that "tickless is enabled by default for bare metal" but rather that tickless is a RTOS optimisation, it has no meaning in bare metal.

Reply:
@rajkan01 Look at the documentation for the Mbed OS Timer: OsTimer

This provides the marshalling point for a system global SysTimer, which is used to provide:
- timed sleeps (for default idle hook in RTOS tickless mode, or non-RTOS sleeps)
- regular ticks for RTOS
- absolute system timing (directly for non-RTOS, or indirectly via RTOS tick count)

Bare metal uses SysTimer to implement timed sleeps. This is a different matter.